### ISSUE TITLE
Reactive Eval::get should propagate errors

### ISSUE THREAD

**Issue by johnmcclean** [aol/cyclops#989]
Currently, for reactive/ CompletableEvals Eval get() swallows any errors recieved via the reactive-streams error channel (but propagates errors thrown inside it's own operators). It returns null instead.

This is a bug. It is reasonable to expect that round triping from a Future with a NoSuchElementException results in a failed Future with a NullPointerException and not a successful Future with a null value.

Eval::get should throw any errors recieved from any Publisher to which it was subscribed (this does make mixing Eval with reactive-streams operators fromPublisher / mergeMap unsafe, likely for the duration of the 10.X.X cycle).

**Comment by johnmcclean** [comment]
Tests

```java
@Test(expected = NoSuchElementException.class)
     public void fromFuture(){
      Future<Integer> f = Future.ofError(new NoSuchElementException());

      Eval.fromPublisher(f).get();


    }

    @Test
    public void toTry(){
        Future<Integer> f = Future.ofError(new NoSuchElementException());
        Try<Integer, Throwable> t = Eval.fromPublisher(f).toTry();
        System.out.println(t);
        assertThat(t.isFailure(),equalTo(true));
        assertThat(t.failureGet().orElse(null),instanceOf(NoSuchElementException.class));
    }
    @Test
    public void fromFuture2(){
        AtomicReference<Throwable> error = new AtomicReference<>(null);
        Future<Integer> f = Future.ofError(new NoSuchElementException());
        Future.fromPublisher(Eval.fromPublisher(f)).recover(e->{
            error.set(e.getCause());return -1;
        });

        assertThat(error.get(),instanceOf(NoSuchElementException.class));
    }
```

**Comment by johnmcclean** [comment]
Released in 10.1.1
